File: backend/src/main/java/outcobra/server/model/MarkReportEntry.java
package outcobra.server.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import lombok.*;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class MarkReportEntry {
    @Id
    private Long id;

    @NotNull
    private Double weight;

    @NotNull
    @ManyToOne
    private Subject subject;

    @NotNull
    @ManyToOne
    private MarkReport report;
}

File: backend/src/main/kotlin/outcobra/server/model/MarkReportEntry.kt
package outcobra.server.model


import outcobra.server.model.interfaces.ParentLinked
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.validation.constraints.NotNull

@Entity
data class MarkReportEntry(@NotNull var weight: Double? = null,
                           @NotNull @ManyToOne var subject: Subject? = null,
                           @NotNull @ManyToOne var report: MarkReport? = null) : ParentLinked, AbstractEntity() {
    override val parent: ParentLinked?
        get() = subject
    //endregion


}
